### PR TITLE
Auto update of components

### PR DESCRIPTION
**Pull request by sheerun**

### CODE REVIEW

File: app/workers/update_scheduler.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class UpdateScheduler
+  include Sidekiq::Worker
+  include Sidetiq::Schedulable
+
+  sidekiq_options :queue => 'update_scheduler'
+
+  recurrence { hourly }
+
+  def perform
+    Component.all.each do |component|
```
Comment (teamon) [inline app/workers/update_scheduler.rb]: http://api.rubyonrails.org/classes/ActiveRecord/Batches.html

File: app/workers/build_version.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class BuildVersion
+  include Sidekiq::Worker
+
+  sidekiq_options queue: 'default', unique: true
+
+  def perform(name, version)
+    puts "Building #{name}##{version}..."
```
Comment (teamon) [inline app/workers/build_version.rb]: `Rails.logger` ?

File: config/application.rb [hunk 3]
```diff
@@ -11,6 +11,8 @@
 # you've limited to :test, :development, or :production.
 Bundler.require(:default, Rails.env)
 
+require_relative '../app/workers/update_scheduler'
```
Comment (teamon) [inline config/application.rb:14]: Why?

File: config/routes.rb [hunk 4]
```diff
@@ -6,6 +6,15 @@
 
   get "/api/v1/dependencies", to: "main#dependencies"
 
+  require 'sidekiq/web'
+  require 'sidetiq/web'
```
Comment (teamon) [inline config/routes.rb:10]: Duplicate

Comment (sheerun) [inline config/routes.rb:10]: side _tiq_

File: app/workers/update_scheduler.rb (hunk 1 above)
Comment (sheerun) [inline app/workers/update_scheduler.rb]: No need for batches. This is only for performing async jobs.

File: config/routes.rb (hunk 4 above)
Comment (teamon) [inline config/routes.rb:10]: Meh, my bad

File: config/application.rb (hunk 3 above)
Comment (sheerun) [inline config/application.rb:14]: Scheduled workers need to be required at startup, or they won't run.

File: app/workers/update_scheduler.rb (hunk 1 above)
Comment (sheerun) [inline app/workers/update_scheduler.rb]: Fixed in e71e5f3

File: app/workers/build_version.rb (hunk 2 above)
Comment (sheerun) [inline app/workers/build_version.rb]: Fixed in 67bc2e2

**Comment by porada** [comment]
@teamon?

**Comment by sheerun** [comment]
Finally

**Pull request merged**